connectors/sources/jira.py

```diff
@@ -0,0 +1,496 @@
+            logger.warning(f"{attachment_name} is not supported by TIKA, skipping")
+            return
+
+        if attachment_size > FILE_SIZE_LIMIT:
+            logger.warning(
+                f"File size {attachment_size} of file {attachment_name} is larger than {FILE_SIZE_LIMIT} bytes. Discarding file content"
+            )
+            return
+
+        logger.debug(f"Downloading {attachment_name}")
+
+        document = {
+            "_id": f"{issue_key}-{attachment['id']}",
+            "_timestamp": attachment["created"],
+        }
+        temp_filename = ""
+        attachment_url = ATTACHMENT_CLOUD if self.is_cloud else ATTACHMENT_SERVER
+        async with NamedTemporaryFile(mode="wb", delete=False) as async_buffer:
+            async for response in self._api_call(
+                url_name=attachment_url,
+                attachment_id=attachment["id"],
+                attachment_name=attachment["filename"],
+            ):
+                async for data in response.content.iter_chunked(CHUNK_SIZE):
+                    await async_buffer.write(data)
+                temp_filename = str(async_buffer.name)
+
+        logger.debug(f"Calling convert_to_b64 for file : {attachment_name}")
+        await asyncio.to_thread(convert_to_b64, source=temp_filename)
+        async with aiofiles.open(file=temp_filename, mode="r") as async_buffer:
+            # base64 on macOS will add a EOL, so we strip() here
+            document["_attachment"] = (await async_buffer.read()).strip()
+        try:
+            await remove(temp_filename)
+        except Exception as exception:
+            logger.warning(
+                f"Could not remove file: {temp_filename}. Error: {exception}"
+            )
+        return document
+
+    async def ping(self):
+        """Verify the connection with Jira"""
+        if self.session is None:
+            self._generate_session()
+
+        if self.ssl_enabled:
+            self.ssl_ctx = ssl_context(certificate=self.certificate)
+        try:
+            await anext(self._api_call(url_name=PING))
+            logger.debug("Successfully connected to the Jira")
+        except Exception:
+            logger.exception("Error while connecting to the Jira")
+            raise
+
+    async def _get_projects(self):
+        """Get projects with the help of REST APIs
+
+        Yields:
+            project: Project document to get indexed
+        """
+        async for response in self._api_call(url_name=PING):
+            timezone = await response.json()
+            self.timezone = timezone["timeZone"]
+
+        async for response in self._api_call(url_name=PROJECT):
+            response = await response.json()
+            for project in response:
+                yield {
+                    "_id": f"{project['name']}-{project['id']}",
+                    "_timestamp": iso_utc(
+                        when=datetime.now(pytz.timezone(self.timezone))
```

Comment:
why do you feed `when` ? it looks like the default value as it's done in `iso_utc`

Reply:
Since, we're indexing project every time with _timestamp with `iso_utc` and other jira documents are using the timestamp getting from the response. So, if Jira server is hosted in different timezone other than UTC, user will face timestamp mismatch for projects and other objects in term of timezone. So, keep it consitent, we're feeding `when` to the `iso_utc` of the timezone of Jira server.